process/sync/metablock.go

```diff
@@ -125,74 +125,178 @@ func (boot *MetaBootstrap) syncFromStorer(
 	blockUnit dataRetriever.UnitType,
 	hdrNonceHashDataUnit dataRetriever.UnitType,
 	notarizedBlockFinality uint64,
-	notarizedHdrNonceHashDataUnit dataRetriever.UnitType,
 ) error {
-	err := boot.loadBlocks(blockFinality,
+	err := boot.loadBlocks(
+		blockFinality,
 		blockUnit,
 		hdrNonceHashDataUnit,
-		boot.getHeaderFromStorage,
+		boot.getMetaHeaderFromStorage,
 		boot.getBlockBody,
-		boot.removeBlockBody)
+		boot.removeBlockBody,
+		boot.getNonceWithLastNotarized,
+		boot.applyNotarizedBlocks,
+		boot.cleanupNotarizedStorage)
 	if err != nil {
 		return err
 	}
 
-	for i := uint32(0); i < boot.shardCoordinator.NumberOfShards(); i++ {
-		err = boot.loadNotarizedBlocks(notarizedBlockFinality,
-			notarizedHdrNonceHashDataUnit+dataRetriever.UnitType(i),
-			boot.applyNotarizedBlock)
+	return nil
+}
+
+func (boot *MetaBootstrap) getNonceWithLastNotarized(nonce uint64) (uint64, map[uint32]uint64, map[uint32]uint64) {
```

Comment:
func too long, might consider splitting it

Reply:
Done